django_project/stakeholder/views.py

```diff
@@ -68,3 +88,376 @@ def get_context_data(self, **kwargs):
+                notifications.append(reminder)
+        return notifications
+    else:
+        return reminders
+
+
+def delete_reminder_and_notification(request):
+    data = json.loads(request.POST.get('ids'))
+    notifications_page = request.POST.get('notifications_page')
+    organisation = request.session[CURRENT_ORGANISATION_ID_KEY]
+    try:
+        for element in data:
+            if isinstance(element, str) and element.isdigit():
+                reminder = Reminders.objects.get(
+                    user=request.user,
+                    organisation=organisation,
+                    id=int(element)
+                )
+                if reminder.user == request.user:
+                    Reminders.objects.filter(
+                        user=request.user,
+                        organisation=organisation,
+                        id=int(element)
+                    ).delete()
+
+        reminders = Reminders.objects.filter(
+            user=request.user,
+            organisation=organisation
+        )
+        if notifications_page is not None:
+            notifications = []
+            for reminder in reminders:
+                if reminder.status == Reminders.PASSED and reminder.email_sent:
+                    notifications.append(reminder)
+            return notifications
+        return reminders
+    except Exception as e:
+        return str(e)
+
+
+def get_reminder_or_notification(request):
+    data = json.loads(request.POST.get('ids'))
+
+    try:
+        for element in data:
+            if isinstance(element, str) and element.isdigit():
+                reminder = Reminders.objects.filter(
+                    user=request.user,
+                    organisation=request.session[CURRENT_ORGANISATION_ID_KEY],
+                    id=int(element)
+                )
+                reminder[0].date = convert_date_to_local_time(reminder[0].date)
+
+        return reminder
+    except Exception as e:
+        return str(e)
+
+
+def paginate(*args):
+    rows = args[0]
+    rows_per_page = args[1]
+    page = args[2]
+    paginator = Paginator(rows, rows_per_page)
+
+    try:
+        paginated_rows = paginator.page(page)
+    except PageNotAnInteger:
+        paginated_rows = paginator.page(1)
+    except EmptyPage:
+        paginated_rows = paginator.page(paginator.num_pages)
+
+    return paginated_rows
+
+
+def get_organisation_reminders(request):
+
+    reminders = Reminders.objects.filter(
+        user=request.user,
+        organisation=request.session[CURRENT_ORGANISATION_ID_KEY]
+    )
+
+    return reminders
+
+
+def adjust_date_to_server_time(request):
+
+    datetime_str = request.POST.get('date')
+
+    # Parse the date string into a datetime object
+    datetime_format = '%Y-%m-%dT%H:%M'
+    parsed_datetime = datetime.strptime(datetime_str, datetime_format)
+
+    # Convert parsed datetime to local timezone (Africa/Johannesburg)
+    local_timezone = pytz.timezone('Africa/Johannesburg')
```

Comment:
@DONOTFORGET-BYTE why do you convert to the South Africa timezone only? What if the user is from a different region? Perhaps you can handle the conversion on the frontend, as the browser will be aware of the user's timezone.